src/globus_action_provider_tools/authentication.py

```diff
@@ -317,9 +331,12 @@ def __init__(
         self,
         auth_client: globus_sdk.ConfidentialAppAuthClient,
         expected_scopes: Iterable[str],
+        *,
+        client_factory: ClientFactory | None = None,
     ) -> None:
         self.auth_client = auth_client
         self.default_expected_scopes = frozenset(expected_scopes)
+        self.client_factory = client_factory
```

Comment:
Does this need to instantiate `ClientFactory` if `client_factory` is `None`?

Reply:
It was intentionally omitted since it's not strictly necessary to create it here (since it gets handled in `AuthState.__init__`), but now that you've asked, I see two reasons to do it here:
- more efficient (instantiate once rather than N times)
- narrows `AuthStateBuilder.client_factory` on instances from `ClientFactory | None` to `ClientFactory`

So I'll change that.